### PR TITLE
feat: support tcp socket & remove udp communicate method with p2p

### PR DESCRIPTION
**Pull request by ai-chen2050**

### CODE REVIEW

File: zchronod/zchronod/src/bin/client_read.rs [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 use protos::{bussiness::{GatewayType, QueryByMsgId, QueryByTableKeyId, QueryMethod, QueryResponse, ZGateway}, innermsg::{Action, Identity, Innermsg}, zmessage::{ZMessage, ZType}};
+use std::io::{Read, Write};
 
 fn main() -> std::io::Result<()> {
-    let socket = UdpSocket::bind("127.0.0.1:34000")
-        .expect("couldn't bind to address");
-
+    let destination = "127.0.0.1:8050";
+    let mut stream = TcpStream::connect(destination)?;
+
```
Comment (bufrr) [inline zchronod/zchronod/src/bin/client_read.rs]: Need to format code using 'rustfmt'

File: zchronod/zchronod/src/bin/client_read.rs [hunk 2]
```diff
@@ -20,24 +21,19 @@ fn main() -> std::io::Result<()> {
...
-            let ret = QueryResponse::decode(response.message.unwrap().data.as_ref()).unwrap();
-            println!("Received response: {:?}", ret);
-        }
-        Err(ref err) if err.kind() == std::io::ErrorKind::WouldBlock => {
-            println!("No response received.");
-        }
-        Err(err) => {
-            eprintln!("Error receiving response: {}", err);
-        }
+    let size = stream.read(&mut buf)?;
```
Comment (bufrr) [inline zchronod/zchronod/src/bin/client_read.rs]: You may need to do `stream.read` for multiple times if client send bytes larger than TCP MTU. So we need to define the transport protocol first.

File: zchronod/zchronod/src/zchronod.rs [hunk 3]
```diff
@@ -81,32 +81,52 @@ impl ServerState {
...
-        let msg = prost::bytes::Bytes::copy_from_slice(&buf[..n]);
-        if let Ok(m) = Innermsg::decode(msg) {
-            handle_msg(arc_zchronod.clone(), m, src).await;
-        } else {
-            println!("No action, only support innermsg type between vlc & p2p modules at now");
-        }
+        let (mut socket, _) = arc_zchronod.read().await.listener.accept().await.unwrap();
+        let arc_zchronod_copy = Arc::clone(&arc_zchronod);
+        tools::tokio_zchronod::block_forever_on(async move {
+            let mut buffer = [0; 65535];
```
Comment (bufrr) [inline zchronod/zchronod/src/zchronod.rs]: Make buffer out of loop

**Review by bufrr — CHANGES_REQUESTED** [review]

File: zchronod/zchronod/src/zchronod.rs (hunk 3 above)
Comment (ai-chen2050) [reply zchronod/zchronod/src/zchronod.rs]: one connection, one buffer

File: zchronod/zchronod/src/bin/client_read.rs (hunk 1 above)
Comment (ai-chen2050) [reply zchronod/zchronod/src/bin/client_read.rs]: sure

File: zchronod/zchronod/src/bin/client_read.rs (hunk 2 above)
Comment (ai-chen2050) [reply zchronod/zchronod/src/bin/client_read.rs]: will refactor when later

**Comment by ai-chen2050** [comment]
The next step will work on it and close the pull request temporarily.

**Pull request closed without merging**